### PR TITLE
[OCPCLOUD-1247] Pass platform type to WMCB 

### PR DESCRIPTION
**Pull request by Fedosin**
This commit passes platform typy to WCMB using the new `--platform-type` parameter.

Depending on the platform, WCMB may perform additional actions to configure the nodes. For instance, on AWS it overrides default hostname for kubelet using values from metadata server.

### CODE REVIEW

File: pkg/csr/csr.go
Comment (JoelSpeed) [inline pkg/csr/csr.go:308]: Is it worth a comment here as to why the `platformType` parameter is empty, something along the lines of `We don't need to pass the platform here because ...`?

File: pkg/nodeconfig/nodeconfig.go
Comment (sebsoto) [inline pkg/nodeconfig/nodeconfig.go:164]: A new nodeconfig object is created every time a VM is configured/deconfigured.
The platform type will never change throughout the lifecycle of the operator so we should avoid finding this more than once.
The platform type is already part of the [instanceReconciler struct](https://github.com/openshift/windows-machine-config-operator/blob/962bd45cc0e7a9735dadc2109d65b57a95d842d2/controllers/controllers.go#L53-L54) and can be passed in when calling `NewNodeConfig`

File: pkg/windows/windows.go
Comment (sebsoto) [inline pkg/windows/windows.go:568]: In what case will `platformType == ""`?

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: pkg/windows/windows.go
Comment (Fedosin) [reply pkg/windows/windows.go:568]: Yeah, previously it could be empty, but now it always has some value. This check is not necessary anymore.

File: pkg/nodeconfig/nodeconfig.go
Comment (Fedosin) [reply pkg/nodeconfig/nodeconfig.go:164]: Done

File: pkg/nodeconfig/nodeconfig.go
Comment (Fedosin) [reply pkg/nodeconfig/nodeconfig.go:24]: Done

File: pkg/csr/csr.go
Comment (Fedosin) [reply pkg/csr/csr.go:308]: Done

File: pkg/nodeconfig/nodeconfig.go
Comment (sebsoto) [inline pkg/nodeconfig/nodeconfig.go:69]: I'm not seeing this used anywhere by nodeconfig in this PR. In that case it should be removed from the struct and only passed into windows.New().

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: pkg/nodeconfig/nodeconfig.go
Comment (sebsoto) [reply pkg/nodeconfig/nodeconfig.go:69]: Yep, while that is the case, as a rule of thumb, I'd prefer to only introduce struct changes in the same PR they are being utilized. For reasons such as plans changing when working on the other PR, and making it easier to revert changes, if that becomes necessary.

With the nodeconfig struct change this PR will be fine to merge after #808 merges, or it can be merged before, with this change removed.

Comment (Fedosin) [reply pkg/nodeconfig/nodeconfig.go]: ack, it's fixed now

**Pull request merged**
